### PR TITLE
RELEASE.md: clarify instructions

### PR DESCRIPTION
**Pull request by simonpasquier**

### CODE REVIEW

File: RELEASE.md [hunk 1]
```diff
@@ -30,31 +30,26 @@ If a bug fix got accidentally merged into master, cherry-pick commits have to be
...
 ```bash
 $ make clean generate
 ```
 
-Bump the version for `pkg/apis/monitoring` package in go.mod:
+Bump the version of the `pkg/apis/monitoring` package in `go.mod`:
 
 ```bash
-$ go get "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring@v${VERSION}"
+$ go mod edit -require "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring@v{VERSION}"
```
Comment (paulfantom) [inline RELEASE.md:52]:
```suggestion
$ go mod edit -require "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring@v$(cat VERSION)"
```

Comment (paulfantom) [reply RELEASE.md:52]: This way it will be a copy-paste type of command

**Review by paulfantom — APPROVED** [review]
overall looks good to me, just one nit

File: RELEASE.md [hunk 2]
```diff
@@ -30,31 +30,26 @@ If a bug fix got accidentally merged into master, cherry-pick commits have to be
 
 Maintaining the release branches for older minor releases happens on a best effort basis.
 
-## Prepare your release
-
-Bump the version in the `VERSION` file in the root of the repository.
+## Update Go dependencies
 
-While on the master branch tag `pkg/apis/monitoring` package:
+A couple of days before the release, consider submitting a PR against the `master` branch to update the Go dependencies.
```
[the remark this answers is not in the document]
Comment (simonpasquier) [reply RELEASE.md:35]: :+1:

**Review by lilic — APPROVED** [review]
lgtm

**Pull request merged**